boards: xtensa: add support for DOIT ESP32 DevKit V1

heronet opened the pull request:
This PR adds support for the DOIT ESP32 DevKit V1, an entry-level
development board based on the ESP32-WROOM-32.

Main features:
- ESP32-WROOM-32 module (4 MB Flash, 520 KB SRAM)
- Dual-core Xtensa LX6 processor up to 240 MHz
- 802.11b/g/n Wi-Fi and Bluetooth v4.2 BR/EDR + BLE
- CP2102 USB-to-UART bridge
- Onboard blue LED connected to GPIO2
- All GPIO pins exposed on 2x15 pin headers

The board has been tested with the following Zephyr samples:
- samples/basic/blinky
- samples/hello_world

heronet commented on tests/drivers/dac/dac_api/src/test_dac.c:
```diff
@@ -74,6 +74,7 @@
 #define DAC_CHANNEL_ID		0
 
 #elif defined(CONFIG_BOARD_ESP32_DEVKITC) || \
+	defined(CONFIG_BOARD_DOIT_ESP32_DEVKIT_V1) || \
```
I actually did it without it first but the twister fails with a unsupported board from the dac test. 

Adding the board to test_dac fixed the issue, at least locally. It fails in this PR I think because the vendor isn't valid yet.

This `west twister -p doit_esp32_devkit_v1/esp32/procpu -s drivers.dac.api` is the test in question

marekmatej replied:
not sure, but the twister issue might be because of "Unsuported vendor". I also think you dont need to do this as we are testing `ESP32` using the `esp32_devkitc` board.

sylvioalves commented:
@heronet please add the proper overlay file in boards folder (dac test), otherwise it wont work.

mbolivar reviewed (commented):
the vendor prefix is pretty short, but I didn't see any conflicts with any stock ticker symbols on yahoo finance. @heronet I don't object to using `doit`, but did you consider using something like `sibo-zhilian`? I'm a little nervous about future-proofing for any zephyr-specific vendor prefix that's 4 characters or less since those might conflict with stock ticker symbols added later.

heronet commented:
> the vendor prefix is pretty short, but I didn't see any conflicts with any stock ticker symbols on yahoo finance. @heronet I don't object to using `doit`, but did you consider using something like `sibo-zhilian`? I'm a little nervous about future-proofing for any zephyr-specific vendor prefix that's 4 characters or less since those might conflict with stock ticker symbols added later.

Considering they have "Copyright © 2024 DOIT" on their website, can another company get it as a stock ticker symbol? I don't know enough about that. But the DOIT esp32 is likely the most used and available esp32 in the wild. It's used in most of the tutorials available online. and `sibo-zhilian` might not be the easiest to find for a new user. 

Since there is no conflicting ticker symbol, maybe we can go with this?

marekmatej commented:
> the vendor prefix is pretty short, but I didn't see any conflicts with any stock ticker symbols on yahoo finance. @heronet I don't object to using `doit`, but did you consider using something like `sibo-zhilian`? I'm a little nervous about future-proofing for any zephyr-specific vendor prefix that's 4 characters or less since those might conflict with stock ticker symbols added later.

@heronet, if there are more boards to come, then introducing a new vendor is fine. Otherwise, consider placing it under the `boards/others` folder.
I'll leave more comments in review.

heronet commented:
> @heronet Thanks for your work. However, I need to ask. Is the presence of an LED on GPIO2 the only reason for adding this board? The board is literally the same as Espressif `ESP32_DevKitC`, to which you can add a simple overlay to be able to use the `blinky` sample. But if you are committed to maintaining this board, I have zero objections.

Thanks for the review. I don't have a espressif esp32 but there are more reasons for having this board than just the LED. Like this board has labeled UART0/2 pins. So I had to create a pinmux different from the espressif esp32 board. This includes other pins as well.

The main idea was convinience. I see this specific board everywhere and it is very popularly known as DOIT ESP32. Seeing it missing in Zephyr I thought adding it would be a good idea.

marekmatej commented:
> > > the vendor prefix is pretty short, but I didn't see any conflicts with any stock ticker symbols on yahoo finance. @heronet I don't object to using `doit`, but did you consider using something like `sibo-zhilian`? I'm a little nervous about future-proofing for any zephyr-specific vendor prefix that's 4 characters or less since those might conflict with stock ticker symbols added later.
> > 
> > 
> > @heronet, if there are more boards to come, then introducing a new vendor is fine. Otherwise, consider placing it under the `boards/others` folder. I'll leave more comments in review.
> 
> I currently don't have access to their other boards. But is there a possibility someone might add one in the future? Considering there are other board vendors with just 1 board at the time in the tree.
> 
> I can put it in others if you think thats the better idea.

In that case I think `others` is a better idea for now.

marekmatej commented on boards/doit/index.rst:
```diff
@@ -0,0 +1,10 @@
+.. _boards-doit:
```
As I mentioned elsewhere, putting this in `others` is a better option and would help to solve the CI issues.

marekmatej reviewed (changes requested):
Overall looks good. A few things to solve:

marekmatej commented on boards/others/esp32_devkit_v1/doc/index.rst:
```diff
@@ -0,0 +1,46 @@
+
+.. include:: ../../../espressif/common/soc-esp32-features.rst
+   :start-after: espressif-soc-esp32-features
+
```
You might want to emphasize the presence of the onboard LED and the GPIO it is connected to. Running the blinky sample is a very frequent question :)

heronet replied:
I added the the LED and a few other info

marekmatej commented on boards/others/esp32_devkit_v1/doit_esp32_devkit_v1_procpu.yaml:
```diff
@@ -0,0 +1,22 @@
+  - uart
+  - nvs
+  - pwm
+  - dac
```
the `dac` is on the list twice. I suggest to remote the `dac` entry altogether to get rid of the `dac_api` issues. Or, you can add the board under the supported in that test, but I don't think it is a good idea for low profile board like this.

marekmatej replied:
ah, ok thanks I'll provide a fix, we still can submit fixes into this release window.

sylvioalves reviewed (changes requested):
I think the folder name could be renamed to `doit_esp32_devkit_v1`. @marekmatej what do you think?

heronet committed:
boards: xtensa: add support for DOIT ESP32 DevKit V1

This patch adds support for the DOIT ESP32 DevKit V1, an entry-level
development board based on the ESP32-WROOM-32.

Main features:
- ESP32-WROOM-32 module (4 MB Flash, 520 KB SRAM)
- Dual-core Xtensa LX6 processor up to 240 MHz
- 802.11b/g/n Wi-Fi and Bluetooth v4.2 BR/EDR + BLE
- CP2102 USB-to-UART bridge
- Onboard blue LED connected to GPIO2
- All GPIO pins exposed on 2x15 pin headers

The board has been tested with the following Zephyr samples:
- samples/basic/blinky
- samples/hello_world

Signed-off-by: Siratul Islam <sirat4757@gmail.com>
